engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/Handler.scala

```diff
@@ -116,15 +119,17 @@ final class Handler {
     */
   def onMessage(request: Api.Request): Unit = request match {
     case Api.Request(requestId, Api.ShutDownRuntimeServer()) =>
-      if (commandProcessor ne null) {
-        commandProcessor.stop()
+      if (ctx != null) {
+        ctx.commandProcessor.stop()
       }
       endpoint.sendToClient(
         Api.Response(requestId, Api.RuntimeServerShutDown())
       )
 
     case request: Api.Request =>
-      val cmd = CommandFactory.createCommand(request)
-      commandProcessor.invoke(cmd)
+      if (ctx != null) {
```

Comment:
The case when the context is not initialized, and the handler already started receiving messages most likely indicates an error somewhere else. We should at least log it appropriately.

Reply:
Yeah, except without having a reference to `TruffleContext` I don't have the logger either. Need to figure out how to pass it around.

Reply:
Throwing an exception. This is an internal error that should always be satisfied